cmd/wego/add/cmd.go

```diff
@@ -36,17 +38,28 @@ func init() {
 	Cmd.Flags().StringVar(&params.Branch, "branch", "main", "Branch to watch within git repository")
 	Cmd.Flags().StringVar(&params.DeploymentType, "deployment-type", "kustomize", "deployment type [kustomize, helm]")
 	Cmd.Flags().StringVar(&params.Chart, "chart", "", "Specify chart for helm source")
+	Cmd.Flags().StringVar(&params.PrivateKey, "private-key", "", "Private key to access git repository over ssh")
 	Cmd.Flags().StringVar(&params.AppConfigUrl, "app-config-url", "", "URL of external repository (if any) which will hold automation manifests; NONE to store only in the cluster")
 	Cmd.Flags().BoolVar(&params.DryRun, "dry-run", false, "If set, 'wego add' will not make any changes to the system; it will just display the actions that would have been taken")
 }
 
 func runCmd(cmd *cobra.Command, args []string) {
 	params.Namespace, _ = cmd.Parent().Flags().GetString("namespace")
 
-	gitClient := git.New(&http.BasicAuth{
-		Username: "weaveworks", // this can't be empty
-		Password: os.Getenv("GITHUB_TOKEN"),
-	})
+	if strings.HasPrefix(params.PrivateKey, "~/") {
+		dir := getHomeDir()
+		params.PrivateKey = filepath.Join(dir, params.PrivateKey[2:])
+	} else if params.PrivateKey == "" {
+		params.PrivateKey = findPrivateKeyFile()
+	}
+
+	authMethod, err := ssh.NewPublicKeysFromFile("git", params.PrivateKey, params.PrivateKeyPass)
```

Comment:
We should add the private-key-pass to the CLI flags.

Reply:
If we decide passing the password on the command line is okay, we can expose it. It has long been standard practice to avoid that since it shows up in /proc. If the assumptions around that have changed we can revisit this.